src/profile-logic/import/linux-perf.js

```diff
@@ -6,19 +6,33 @@
 
 import type { MixedObject } from 'firefox-profiler/types';
 
+/** Whether a line should be ignored by parsing. */
+function isSkippableLine(line: string): boolean {
+  // perf script --header outputs header lines beginning with #
+  return line === '' || line.startsWith('#');
+}
+
 /**
  * The "perf script" format is the plain text format that is output by an
  * invocation of `perf script`, where `perf` is the Linux perf command line tool.
  */
 export function isPerfScriptFormat(profile: string): boolean {
-  const firstLine = profile.substring(0, profile.indexOf('\n'));
-  //          +- process name (anything before the rest of the regexp, can contain spaces)
-  //          |        +- PID/ (optional)
-  //          |        |      +- TID
-  //          |        |      |          +- [CPU] (optional, present in SimplePerf output)
-  //          |        |      |          |          +- timestamp
-  //       vvvvv   vvvvvvvvv vvv   vvvvvvvvvvvvvv vvvvvv
-  return /^\S.+?\s+(?:\d+\/)?\d+\s+(?:\[\d+\]\s+)?[\d.]+:/.test(firstLine);
+  const lines = profile.split('\n');
```

Comment:
You could argue this is too expensive a call to do in a filetype sniffer, perhaps? I hadn't really considered it before pushing.

Reply:
That's a great idea, impressive regex-skills. I haven't used a multiline regex in a long time, and it solves the problem nicely 👍

Reply:
It passes the test I have. It /might/ fail if something outputs whitespace-only lines, but I don't actually have any examples of anything doing that, so I think it's OK to ignore until it becomes a problem.

Reply:
I'm thinking that in the future we'll likely want to slice the input to a reasonable length, because this can also be abused by one very long line. But this was also the case with the current code...

Reply:
Done -- I sliced the input to 20kB, a 10x margin of error over my current profile.

Reply:
I may miss something, as I don't see these new changes :-)

Reply:
Woops. I thought I'd pushed but there was a lint error.